texar/torch/modules/embedders/embedder_base.py

```diff
@@ -142,9 +142,8 @@ def forward(self,  # type: ignore
 
     @property
     def output_size(self) -> int:
-        r"""The final dimension(s) of :meth:`forward` output tensor(s).
-
-        Here final dimension is ``1``, because here output tensor size equals
-        to the input tensor size.
+        r"""The feature size of :meth:`forward` output.
+        Equals to ``1`` because here output size equals
+        to the input size.
         """
         return 1
```

Comment:
I'm actually thinking whether we need `output_size` for this class. We use a size of 1 for classifiers because when `num_classes` is one we can interpret the output as having a size of `[batch_size, 1]` (although we squeeze that final dimension), but it's not the case here and having 1 does not really make sense.
Since no one will probably use this method, I'm thinking maybe we could just throw a `ValueError` for the modules where we can't calculate the output size.

Reply:
Thanks, would you think we remain the function but throw out `ValueError`? In that case how about we use `AttributeError`?

Reply:
It depends on how we want to interpret this error:

- If the error occurs because of certain configurations or values, it's a `ValueError`.
- If the error occurs because the class in question does not support such operations, it's a `TypeError`.
- If the error occurs during an attribute lookup and fails because the instance in question does not have such attribute, it's an `AttributeError`.

In our case, using `AttributeError` might confuse the user. I think it should be either `ValueError` or `TypeError`, maybe `TypeError` is more appropriate here.